src/tools/genpolicy/src/registry.rs

```diff
@@ -215,41 +200,182 @@ impl Container {
     }
 }
 
+async fn get_config_layer(image_ref: String, socket_path: String) ->  Result<DockerConfigLayer, Box<dyn std::error::Error>>{
+    
+    let socket = socket_path.clone(); // todo: figure out how not to clone everything to get it working
+    let channel = Endpoint::try_from("http://[::]")
+        .unwrap()
+        .connect_with_connector(service_fn(move |_: Uri| UnixStream::connect(socket.clone())))
+        .await
+        .expect("Could not create client.");
+
+    let mut client = ImageServiceClient::new(channel);
+
+    let req =   k8s_cri::v1::ImageStatusRequest {
+        image: Some(k8s_cri::v1::ImageSpec {
+            image: image_ref,
+            annotations: HashMap::new(),
+        }),
+        verbose: true
+    };
+
+    let resp = client.image_status(req).await?;
+    let image_layers = resp.into_inner();
+
+    let status_info: serde_json::Value = serde_json::from_str(image_layers.info.get("info").unwrap())?;
+    let image_spec = status_info["imageSpec"].as_object().unwrap();
+    let docker_config_layer: DockerConfigLayer = serde_json::from_value(serde_json::to_value(image_spec)?).unwrap();
+
+    Ok(docker_config_layer)
+}
+
+pub async fn pull_image(image_ref: String, socket_path: String) ->  Result<(), anyhow::Error>{
```

Comment:
Would just Result<()> work here? We can return an error just using Result<()> - unless I am missing something - no need for anyhow:Error.